frontend/packages/console-dynamic-plugin-sdk/src/coderefs/coderef-resolver.ts

```diff
@@ -126,3 +126,33 @@ export const resolveCodeRefProperties = async <E extends Extension<P>, P = Exten
 
   return resolvedValues;
 };
+
+/**
+ *  Returns an extension with its resolved `CodeRef` properties.
+ */
+export const resolveExtensionWithCodeRefs = <E extends Extension<P>, P = ExtensionProperties<E>>(
```

Comment:
```suggestion
export const resolveExtension = async <E extends Extension<P>, P = ExtensionProperties<E>>(
```

Reply:
no need for this. we want to return a promise or an array of promises and let the user deal with the `onFulfil, onError`, isnt it a better approach?